discoverd/client/heartbeat.go

```diff
@@ -131,7 +133,9 @@ func (h *heartbeater) client() *Client {
 }
 
 const (
-	heartbeatInterval        = 5 * time.Second
+	// As a random amount of time of up to half of the heartbeatInterval is
+	// subtracted, it will average to an interval of 5s.
+	heartbeatInterval        = 6666 * time.Millisecond
```

Comment:
Even with the comment, it took me a minute to understand why this was 6⅔ and not 5 (I now realise the heartbeat follows a uniform distribution `X = U(3⅓,  6⅔)`, so thus has an expected value of 5).

I think it would be clearer to make this value the mean value (e.g. `meanHeartbeatInterval = 5 * time.Second`) and add another constant like `maxHeartbeatJitter`, thus making heartbeat interval follow a uniform distribution between `meanHeartbeatInterval ± maxHeartbeatJitter`.

WDYT?